common/src/autogluon/common/utils/resource_utils.py

```diff
@@ -159,9 +159,22 @@ def _get_gpu_count_cuda():
         cudaShutdown()
         return gpu_count
 
+    @staticmethod
+    def _get_custom_memory_size():
+        memory_limit = float(os.environ.get("AG_MEMORY_LIMIT_IN_GB"))
+
+        if memory_limit <= 0:
+            raise ValueError("Memory set via `AG_MEMORY_LIMIT_IN_GB` must be greater than 0!")
+
+        # Transform to bytes and return
+        return max(int(memory_limit * (1024.0**3)), 1)
```

Comment:
Note: If AutoGluon's memory estimates are off for a model, and the model uses more memory than it is supposed to, this could still lead to memory over-allocation in the SLURM cluster. Unsure if there is a good work-around for this besides "don't incorrectly estimate memory usage", maybe the wrapper logic that errors a function if using too much memory via `pynisher`, but I suspect that has a performance tradeoff.

Reply:
I wonder if Ray has any logic when creating a worker to limit memory usage of the worker? Couldn't find anything from a quick google search.

For example, Ray kills workers if >95% memory usage, but probably would ignore the memory specified in this PR's patch: https://docs.ray.io/en/latest/ray-core/scheduling/ray-oom-prevention.html

Reply:
That case would also error/kill a normal run without SLURM. Which is then caught, and the model is simply disregarded as failed (as only the fitting process is killed if the ray is used). 

One can adjust Ray's soft and hard limits. However, we would likely need to monkey patch its memory estimations via psutil again. I would be interested to see how far one could allocate memory with Ray jobs, too. In theory, it could have the same API as CPUs/GPUs, but I am not aware that it is supported so far.